view/frontend/templates/js/replacejs.phtml

```diff
@@ -588,6 +614,8 @@ ob_start();
                     }
                 }
 
+                if (!cart.orderToken) return false;
```

Comment:
@daisy1754 this line fixes the entitled issue. However, when testing on a slow network I had a couple of cases that connect.js did not load on time. It is loaded asynchronously. So I added a fix for that as well in the same PR cause I thought this was kind of urgent.

Reply:
can you open issue in engbugs with repro steps? I think it's better to handle this in connect.js side

said above, it may take a while to change connect.js - if we want to get change around waiting BoltCheckout let's do it as a separate PR

regarding this line, as Mike mentioned in asana I think intended way of doing is you pass cart promise in BoltCheckout.configure so we can wait. can you remind me why we cannot do that? (might be better to reply in asana so Mike can see your answer as well)

Reply:
@daisy1754 we do pass cart as a promise when calling BoltCheckout.configure initially and on every cart change - in createOrder function. However, there are cases when only hints  change and we don't call crating of a new cart. We collect hints prefill from the front end - country, state, zip on the shopping cart page (there's support for the complete address on the checkout page that's not actively used). When hints change we call BoltCheckout.configure(cart, hints, callbacks) with the current state of the cart. If cart hasn't been fetched yet (slow network) we get the undefined cart error. This was the problem. It's fixed now.
cc @matt-thomason

Reply:
sorry for delay - if you just want to update hint and not want to mess with cart you can pass null as argument

Reply:
or you can hold reference to cart promise and keep reusing it

Reply:
Thaks @daisy1754 that could help a lot.

Reply:
Passing null is the cause of the issue. One of a few. connect.js loads and button is placed on the page before the javascript executes (all happen independently, asynchronously). The solution to this, at least from plugin perspective, is to prevent the checkout to pop up (callbacks.check returns false) until the javascript loads and reconfigures it. Trade off is a potential small number of dead clicks. Fixed the hints update by reusing the cart promise, great tip by Kazuki.